datadog/resource_datadog_security_monitoring_rule.go

```diff
@@ -510,15 +521,15 @@ func resourceDatadogSecurityMonitoringRuleCreate(ctx context.Context, d *schema.
 	return nil
 }
 
-func isSignalCorrelationSchema(d *schema.ResourceData) bool {
+func isSignalCorrelationSchema(d utils.Resource) bool {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏Why did you have to update the method signature?

Reply:
The method is called from `buildCreatePayload` and it is used from multiple places. One where we want to create payload from `ResourceData`, and other where we want to create from `ResourceDiff`. Since they are separate (and not one extending other), I found `utils.Resource` being a parent interface. With that, we can pass either `ResourceData/ResourceDiff` and it accepts them.